packages/upload-client/src/blob.js

```diff
@@ -280,32 +280,36 @@ export async function add(
   }
 
   // Invoke `conclude` with `http/put` receipt
-  const derivedSigner = ed25519.from(
-    /** @type {import('@ucanto/interface').SignerArchive<import('@ucanto/interface').DID, typeof ed25519.signatureCode>} */
-    (nextTasks.put.task.facts[0]['keys'])
-  )
-
-  const httpPutReceipt = await Receipt.issue({
-    issuer: derivedSigner,
-    ran: nextTasks.put.task.cid,
-    result: { ok: {} },
-  })
+  let { receipt: httpPutReceipt } = nextTasks.put
+  if (!httpPutReceipt?.out.ok) {
+    const derivedSigner = ed25519.from(
+      /** @type {import('@ucanto/interface').SignerArchive<import('@ucanto/interface').DID, typeof ed25519.signatureCode>} */
+      (nextTasks.put.task.facts[0]['keys'])
+    )
+    httpPutReceipt = await Receipt.issue({
+      issuer: derivedSigner,
+      ran: nextTasks.put.task.cid,
+      result: { ok: {} },
+    })
+  }
   const httpPutConcludeInvocation = createConcludeInvocation(
     issuer,
     // @ts-expect-error object of type unknown
     audience,
     httpPutReceipt
   )
   const ucanConclude = await httpPutConcludeInvocation.execute(conn)
```

Comment:
I am thinking this call should only be executed if we're missing a http/put receipt. however, I'm leaving it here as a "validation" the receipt we got makes sense.

Reply:
hmm, I think we don't need to conclude if we already have the receipt for a concluded task.

@vasco-santos is that right?

Reply:
addressed